code/krepel/math/math.d

```diff
@@ -0,0 +1,82 @@
+module krepel.math.math;
+
+import std.math;
+import krepel.algorithm.comparison;
+
+@safe:
+@nogc:
+
+/// Returns the square root of the given value
+float Sqrt(float value)
+{
+  // TODO replace with own (opcode?)
+  return std.math.sqrt(value);
+}
+
+/// Returns the absolute value (the positive value)
+float Abs(float value)
+{
+  // TODO replace with own (opcode?)
+  return std.math.abs(value);
+}
+
+/// Checks if the given float value is QNaN
+bool IsNaN(float value)
+{
+  bool result;
+  asm @nogc @safe
+  {
+    fld value; // Load value to float stack ST(0)
+    ftst; // Compare ST(0) with 0.0
+    fstsw AX; // Write FPU Status Register to AX (16 Bit)
+    and AX, 0x4500; // Filter interesting bits (C0 C2 C3)
+    cmp AX, 0x4500; // Check if C0 C2 and C3 are set
+    lahf; // Load result of cmp into AH
+    and AH, 0x41; // Filter intereseting bits (CF, ZF)
+    xor AH, 0x40; // Check if ZF is set and CF is zero.
+    mov result, AH; // Write result into bool variable
+  }
+  return !result;
+}
+
+/// Checks if a and b are equals with respect to some epsilon
+/// Epsilon gets scaled by the bigger of the two values to compare
+/// to compensate the reduced precision when using higher magnitudes
+bool NearlyEquals(float a, float b, float epsilon = 1.0e-4f)
```

Comment:
Do we want a global constant that defines the default epsilon? Such as `enum KindaSmallNumber = 1.0e-4f;`? Or like in ezEngine with something like this:

```
enum : float
{
  HugeEpsilon    = 1.0e-3f,
  LargeEpsilon   = 1.0e-4f,
  DefaultEpsilon = 1.0e-5f,
  SmallEpsilon   = 1.0e-6f,
}
```

I'm fine with hard-coding the epsilon though.